### ISSUE TITLE
Update the documents

### ISSUE THREAD

**Issue by rawhuul** [iced-rs/iced#2515]
### Is your issue REALLY a bug?

- [X] My issue is indeed a bug!
- [X] I am not crazy! I will not fill out this form just to ask a question or request a feature. Pinky promise.

### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Is this issue related to iced?

- [X] My hardware is compatible and my graphics drivers are up-to-date.

### What happened?

The [doc.rs](https://docs.rs/iced/latest/iced/widget/trait.Component.html) page shows their is `Component` trait in `iced::widget` module, however it is not present in module and even getting docs from `cargo doc --open`, doesn't show the trait.

My `Cargo.toml` is:
```toml
[package]
name = "abc"
version = "0.1.0"
edition = "2021"

[dependencies]
iced = { version = "0.12.1", features = ["tokio", "advanced"] }
tokio = { version = "1.38.1", features = ["full"] }
```

### What is the expected behavior?

Is `Container` trait deprecated? If not, in which module it is located?

### Version

crates.io release

### Operating System

Windows

**Comment by zabackary** [comment]
The `Component` trait is [behind the Cargo feature `lazy`](https://docs.rs/iced_widget/0.12.1/src/iced_widget/lib.rs.html#52-53), which is why it's not in your local `cargo docs`-generated documentation. It appears on `crates.io` because `crates.io` is configured specially to show documentation for all features.

It also confused me a bit when I first used `Component`, so it might be good to update that docs. I'm not sure why docs.rs doesn't put a <kbd>lazy</kbd> badge next to that trait. It's possible it's because the feature gate is actually located in the child package `iced-widget`.

**Comment by hecrj** [comment]
Fixed by #2289.
